x/oracle/keeper/keeper.go

```diff
@@ -80,10 +80,33 @@ func (k Keeper) GetExchangeRate(ctx sdk.Context, denom string) (sdk.Dec, error)
 		return sdk.ZeroDec(), sdkerrors.Wrap(types.ErrUnknownDenom, denom)
 	}
 
-	dp := sdk.DecProto{}
-	k.cdc.MustUnmarshal(b, &dp)
+	decProto := sdk.DecProto{}
+	k.cdc.MustUnmarshal(b, &decProto)
 
-	return dp.Dec, nil
+	return decProto.Dec, nil
+}
+
+// GetExchangeRateBase gets the consensus exchange rate of an asset
+// in the base denom (e.g. ATOM -> uatom)
+func (k Keeper) GetExchangeRateBase(ctx sdk.Context, denom string) (sdk.Dec, error) {
```

Comment:
So we gotta be careful with this method. The caller, `x/leverage`, will only have the base denom, e.g. `uumee` or `ibc/XXXX...` (uatom), so the `denom` argument here will be the base denom.

What denom do we use for storing exchange rates via `GetExchangeRateKey`? The base or symbol? If it's the symbol (my guess), then we'll need to have a reverse index from base -> symbol. Then we use that value for the loop check below.

Reply:
We're using the symbol - we could use the whitelist to get to the base, or add a map to the store when we do ballot counting